### PR TITLE
Add rename mailbox option

### PR DESCRIPTION
**Pull request by GretaD**
Fixes https://github.com/nextcloud/mail/issues/3058

### CODE REVIEW

**Comment by ChristophWurst** [comment]
What was the status here? Please add this info to the description of your PRs so others get a better understanding.

If everything is done from your side please let me know and I'll rebase to latest master.

File: src/components/NavigationMailbox.vue [hunk 1]
```diff
@@ -317,6 +333,34 @@ export default {
 				}
 			)
 		},
+		async renameMailbox(event) {
+			this.renameInput = false
+			this.showSaving = false
+
+			const newName = event.target.querySelector('input[type=text]').value
```
Comment (ChristophWurst) [inline src/components/NavigationMailbox.vue:340]: can we use two-way binding instead to a `data` prop? https://vuejs.org/v2/guide/forms.html

File: src/service/MailboxService.js [hunk 2]
```diff
@@ -46,3 +46,12 @@ export const deleteMailbox = async(id) => {
 
 	await axios.delete(url)
 }
+export async function patchMailbox(accountId, mailboxId, data) {
+	const url = generateUrl('/apps/mail/api/accounts/{accountId}/mailboxes/{mailboxId}', {
```
Comment (ChristophWurst) [inline src/service/MailboxService.js:50]:
```suggestion
	const url = generateUrl('/apps/mail/api/mailboxes/{mailboxId}', {
```

not 100% sure but I think you're using an outdated URL format. Check `appinfo/routes.php` for comparison.

File: src/store/mutations.js [hunk 3]
```diff
@@ -218,4 +218,19 @@ export default {
 	deleteAlias(state, { account, alias }) {
 		account.aliases.splice(account.aliases.indexOf(alias), 1)
 	},
+	renameMailbox(state, { accountId, mailbox, newName }) {
+		const oldId = mailbox.id
+
+		// Remove the old entry
+		Vue.delete(state.mailboxes, mailbox.databaseId(accountId, oldId))
```
Comment (ChristophWurst) [inline src/store/mutations.js:225]: `mailbox.databaseId` is an `int`. You're using it as function.

File: src/store/mutations.js [hunk 4]
```diff
@@ -218,4 +218,19 @@ export default {
...
+		// Update the ID
+		mailbox.id = btoa(newName)
```
Comment (ChristophWurst) [inline src/store/mutations.js:227]: we don't need bas64 encoding anymore

File: src/store/mutations.js [hunk 5]
```diff
@@ -218,4 +218,19 @@ export default {
...
+		// Set the new entry
+		Vue.set(state.mailbox, mailbox.databaseId(accountId, mailbox.id), mailbox)
+
+		// Update the reference
+		const oldNormalized = mailbox.databaseId(accountId, oldId)
```
Comment (ChristophWurst) [inline src/store/mutations.js:232]: again, this makes no sense

File: src/store/mutations.js [hunk 6]
```diff
@@ -218,4 +218,19 @@ export default {
...
+		state.accounts[accountId].mailboxes = state.accounts[accountId].mailboxes.filter(id => id !== oldNormalized)
+		state.accounts[accountId].mailboxes.push(mailbox.databaseId(accountId, mailbox.id))
```
Comment (ChristophWurst) [inline src/store/mutations.js:234]: same here

File: src/store/mutations.js (hunk 3 above)
Comment (GretaD) [reply src/store/mutations.js:225]: oh right, i just copy paste to replace normalizedFolderId

File: src/components/NavigationMailbox.vue [hunk 7]
```diff
@@ -43,21 +43,31 @@
 				</ActionText>
 
 				<ActionButton
-					v-if="mailbox.specialRole !== 'flagged' && !account.isUnified"
```
Comment (ChristophWurst) [inline src/components/NavigationMailbox.vue:46]: was this intentional?

**Review by ChristophWurst — APPROVED** [review]

File: src/components/NavigationMailbox.vue (hunk 7 above)
Comment (GretaD) [reply src/components/NavigationMailbox.vue]: no, fixed it

**Pull request merged**